src/main/java/org/aludratest/service/jms/data/JmsMessageData.java

```diff
@@ -40,13 +40,15 @@
 	}
 
 	public Map<String, Object> getProperties() {
-		return properties;
+		
+		HashMap<String, Object> returnPropList = new HashMap<String, Object>(properties.size());
+		for(Map.Entry<String, Object> entry: properties.entrySet()) {
+			returnPropList.put(entry.getKey(), entry.getValue());
+		}
+		
+		return returnPropList;
 	}
 	
-	public void setProperties(Map<String, Object> props) {
-		this.properties = props;
-	}
-
 	/**
```

Comment:
So how can I remove a property now? Also, you could keep that method with these lines: `this.properties.clear(); this.properties.putAll(props);`